test/false_or_by_contra.lean

```diff
@@ -18,7 +18,7 @@ example (h : False) : ¬ P := by
   guard_target = False
   simp_all
 
-example (h : False) : P := by
+example {P  : Prop} (h : False) : P := by
```

Comment:
?

Reply:
I've made the test clearer. `false_or_by_contra` behaves differently when the goal is a Type vs a Prop. Previously when `P` was an autoimplicit, it was being specialised to `Prop` *by the tactic* in the proof.